### PR TITLE
Add the "Check if an image has loaded"  example

### PR DESCRIPTION
**Pull request by DIRECTcut**
Use Request Logger to log REST API calls: https://stackoverflow.com/questions/52745697/requestlogger-not-intercepting-all-http-requests
from #2

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -47,6 +47,17 @@ http
                 res.end(resultHtml);
             });
         }
+
+        else if (req.url === '/images/image.png') {
```
Comment (miherlosev) [inline server/index.js:51]: This code is not necessary. Remove it.
Create an image with black square and add it to the `server/images` folder.

Rewrite the code below in the following way:
```js
// add to the top of file
const url         = require('url');
const querystring = require('querystring');
....

const repositoryRoot = path.resolve(__dirname, '..');
const resourcePath   = path.join(repositoryRoot, req.url);
const content        = fs.existsSync(resourcePath) ? fs.readFileSync(resourcePath).toString() : '';
const contentType    = CONTENT_TYPES[path.extname(resourcePath)];
const queryStr       = url.parse(req.url).query;
const delay          = parseInt(querystring.parse(queryStr).delay || 0);

if (contentType)
    res.setHeader('content-type', contentType);

 setTimeout(() => {
   res.end(content);
 }, delay);
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-image-loaded-without-errors/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Example page</title>
+</head>
+<body>
+    <div>
+        <img data-testid="img" src="http://localhost:3000/images/image.png">
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/check-if-image-loaded-without-errors/index.html:10]: We recommend attributes like this in the upcoming [best practices article](https://github.com/DevExpress/testcafe/pull/5603/files#diff-cdbad1b185bbda784d95801a61888479R294).
If you disagree, please, leave a comment there.

File: examples/check-if-image-loaded/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Check If an Image Has Loaded Without Errors
```
Comment (miherlosev) [inline examples/check-if-image-loaded/README.md:1]: `# Check If an Image Has Loaded`

Comment (miherlosev) [reply examples/check-if-image-loaded/README.md:1]: `Loaded Without Errors`. 

It's a tautology.

If image is loaded when there is no errors are occurfed during its loading.
Remove `without errors` below.

File: examples/check-if-image-loaded/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+**Test Code**: [index.js](index.js)
+
+**Tested Page**: [index.html](index.html)
+
+This example shows how to check whether an image on the page has loaded without errors.
+
+The tested page requests an image from the server. The server responds with the image after a timeout specified in the request options.
```
Comment (miherlosev) [inline examples/check-if-image-loaded/README.md:9]: The server responds with the image after a specified timeout.

File: examples/check-if-image-loaded-without-errors/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (miherlosev) [reply examples/check-if-image-loaded-without-errors/index.html]: The `data-` attributes are actual only for complex controls with dynamic rendering logic. It's a simplest way to create selectors in such cases.

It tests we don't use such controls. So, using of `data-` attributes are unnecessary complication.

File: server/index.js (hunk 1 above)
Comment (miherlosev) [reply server/index.js]: Add the proposed code to the example.

File: examples/check-if-image-loaded/README.md (hunk 3 above)
Comment (DIRECTcut) [reply examples/check-if-image-loaded/README.md:1]: edited

File: examples/check-if-image-loaded/README.md (hunk 4 above)
Comment (DIRECTcut) [reply examples/check-if-image-loaded/README.md:9]: edited

File: examples/check-if-image-loaded-without-errors/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/check-if-image-loaded-without-errors/index.html]: edited

File: server/index.js (hunk 1 above)
Comment (DIRECTcut) [reply server/index.js]: added

File: examples/check-if-image-loaded/README.md (hunk 3 above)
Comment (DIRECTcut) [reply examples/check-if-image-loaded/README.md:1]: edited

File: examples/check-if-image-loaded/README.md (hunk 4 above)
Comment (DIRECTcut) [reply examples/check-if-image-loaded/README.md:9]: edited

File: examples/check-if-image-loaded-without-errors/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ClientFunction } from 'testcafe';
+
+const checkImageStatus = ClientFunction( () => {
+    const img = document.querySelector('img[data-testid="img"]');
+
+    return new Promise( (resolve, reject) => {
+        img.onload  = () => resolve('Resolved!');
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/check-if-image-loaded-without-errors/index.js]: replaced with funtion(){}

File: examples/check-if-image-loaded/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+**Tested Page**: [index.html](index.html)
+
+This example shows how to check whether an image on the page has loaded.
+
+The tested page requests an image from the server. The server responds with the image after a specified timeout.
+
+If the image loads, the promise is resolved, otherwise it's rejected.
+
+During the test the `t.expect.ok` method checks the promise's status.
```
Comment (miherlosev) [inline examples/check-if-image-loaded/README.md:13]: During the test the `t.expect.ok` method checks the promise's resolved value.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-image-loaded/README.md (hunk 6 above)
Comment (DIRECTcut) [reply examples/check-if-image-loaded/README.md:13]: > During the test, the `t.expect.ok` method waits for the promise to resolve and checks its value.

**Review by miherlosev — APPROVED** [review]

**Review by VasilyStrelyaev — APPROVED** [review]

**Pull request merged**